cloud/services/loadbalancers.go

```diff
@@ -0,0 +1,141 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"github.com/linode/cluster-api-provider-linode/cloud/scope"
+	"github.com/linode/cluster-api-provider-linode/util"
+	"github.com/linode/linodego"
+)
+
+var (
+	defaultLBPort = 6443
+)
+
+// CreateNodeBalancer creates a new NodeBalancer if one doesn't exist
+func CreateNodeBalancer(ctx context.Context, clusterScope *scope.ClusterScope, logger logr.Logger) (*linodego.NodeBalancer, error) {
+	var linodeNBs []linodego.NodeBalancer
+	var linodeNB *linodego.NodeBalancer
+	NBLabel := fmt.Sprintf("%s-api-server", clusterScope.LinodeCluster.Name)
+	clusterUID := string(clusterScope.LinodeCluster.UID)
+	tags := []string{string(clusterScope.LinodeCluster.UID)}
+	filter := map[string]string{
+		"label": NBLabel,
+	}
+
+	rawFilter, err := json.Marshal(filter)
+	if err != nil {
+		return nil, err
+	}
+	logger.Info("Creating NodeBalancer")
+	if linodeNBs, err = clusterScope.LinodeClient.ListNodeBalancers(ctx, linodego.NewListOptions(1, string(rawFilter))); err != nil {
+		logger.Info("Failed to list NodeBalancers", "error", err.Error())
+
+		return nil, err
+	}
+
+	switch len(linodeNBs) {
```

Comment:
this switch and the bottom of this function is hard to follow, e.g. things like reusing `err`, checking `linodeNB != nil` a couple times, default is too many nodebalancers? any chance you'd have motivation to rewrite this with conditionals and returns?
```
if len(linodeNBs) > 1 {
   return nil, errors.New("too many")
}
len(linodeNBs) == 1 {
  // already exists logic
  return &linodeNBs[0], nil  
}

// create logic 

return linodeNB, nil

```

Reply:
This can be simplified even more now that I switched it to find based on `label` since there can only ever be at max one. let me rewrite this logic to be more simple.